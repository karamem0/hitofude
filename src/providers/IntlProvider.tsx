//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
