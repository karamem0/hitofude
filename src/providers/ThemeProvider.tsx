//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import {
  FluentProvider as Provider,
  Theme
} from '@fluentui/react-components';

import { darkTheme, lightTheme } from '../themes/Theme';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';

interface ThemeState {
  theme: Theme,
  themeName: ThemeName
}

interface ThemeContextState {
  theme: ThemeState,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ theme, setTheme ] = React.useState<ThemeState>({
    theme: lightTheme,
    themeName: ThemeName.light
  });

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    switch (value) {
      case ThemeName.light:
        setTheme(() => ({
          theme: lightTheme,
          themeName: value
        }));
        break;
      case ThemeName.dark:
        setTheme(() => ({
          theme: darkTheme,
          themeName: value
        }));
        break;
      default:
        break;
    }
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme,
    changeTheme: handleChangeTheme
  }), [
    theme,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
