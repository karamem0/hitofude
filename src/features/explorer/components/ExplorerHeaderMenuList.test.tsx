//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './ExplorerHeaderMenuList.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';
import { render } from '@testing-library/react';

it('should match the snapshot when the folder is not undefined', () => {
  const params = {
    allFiles: true,
    selectedFolder: {
      id: '01BYE5RZ6QN3ZWBTUFOFD3GSPGOHDJD36K',
      name: 'Attachments',
      parentId: '01BYE5RZ56Y2GOVW7725BZO354PWSELRRZ',
      files: [
        {
          id: '01BYE5RZ4FL7NVUU4UDVHIRG32OLTW4IKM',
          baseName: 'Annual Financial Report (DRAFT)',
          fullName: 'Annual Financial Report (DRAFT).docx',
          extension: '.docx'
        }
      ]
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should match the snapshot when the folder is undefined', () => {
  const params = {
    allFiles: true,
    selectedFolder: undefined
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
