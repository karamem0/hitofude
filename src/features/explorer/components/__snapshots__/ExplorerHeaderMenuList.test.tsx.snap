// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot when the folder is not undefined 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 1rem;
  line-height: 1rem;
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      aria-labelledby=""
      class="fui-MenuList ___1rlijet_kivkug0 f22iagw f1vx9l62 f1t6b6ee"
      data-tabster="{"mover":{"cyclic":true,"direction":1,"memorizeCurrent":true}}"
      role="menu"
    >
      <div
        aria-labelledby="menu-groupr1"
        class="fui-MenuGroup"
        role="group"
      >
        <div
          class="fui-MenuItem rfoezjv"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="fui-MenuItem__icon ro9koqv"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 8a5 5 0 0 1 9-3h-2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 .5-.5v-3a.5.5 0 0 0-1 0v1.53A5.99 5.99 0 0 0 2 8a6 6 0 0 0 11.98.54.5.5 0 0 0-1-.08A5 5 0 0 1 3 8Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span
            class="fui-MenuItem__content r1ls86vo"
          >
            Refresh
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="fui-MenuDivider ___3pxu8i0_ssaivc0 f8dz51a f14z66ap f1facbz3"
        role="presentation"
      />
      <div
        aria-labelledby="menu-groupr2"
        class="fui-MenuGroup"
        role="group"
      >
        <div
          class="fui-MenuItem rfoezjv"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="fui-MenuItem__icon ro9koqv"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m10.12 10.83 4.03 4.02a.5.5 0 0 0 .7-.7l-13-13a.5.5 0 1 0-.7.7l3.23 3.23a6.7 6.7 0 0 0-2.3 3.08l-.05.15-.01.06s-.08.5.35.61a.5.5 0 0 0 .61-.35L3 8.6a3.02 3.02 0 0 1 .2-.52c.16-.34.4-.8.78-1.26.3-.36.66-.72 1.13-1.02l1.57 1.58a2.5 2.5 0 1 0 3.45 3.45Zm-.74-.74a1.5 1.5 0 1 1-1.97-1.97l1.97 1.97Zm-3.06-5.9.85.86C7.43 5.02 7.71 5 8 5c2.04 0 3.29.91 4.03 1.82A5.7 5.7 0 0 1 13 8.6v.02a.5.5 0 0 0 .97-.25v-.02a2.3 2.3 0 0 0-.06-.18 6.7 6.7 0 0 0-1.12-1.98A5.95 5.95 0 0 0 8 4a6.9 6.9 0 0 0-1.68.2Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span
            class="fui-MenuItem__content r1ls86vo"
          >
            Hide other files
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="fui-MenuDivider ___3pxu8i0_ssaivc0 f8dz51a f14z66ap f1facbz3"
        role="presentation"
      />
      <div
        aria-labelledby="menu-groupr3"
        class="fui-MenuGroup"
        role="group"
      >
        <div
          class="fui-MenuItem rfoezjv"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="fui-MenuItem__icon ro9koqv"
          >
            <svg
              aria-hidden="true"
              class="emotion-0 ___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.5 4h1a3.5 3.5 0 0 1 .2 7H9.5a.5.5 0 0 1-.1-.99h.1l1-.01a2.5 2.5 0 0 0 .16-5H9.5a.5.5 0 0 1-.09-1h1.09-1Zm-4 0h1a.5.5 0 0 1 .09 1H5.5a2.5 2.5 0 0 0-.16 5H6.5a.5.5 0 0 1 .09 1H5.5a3.5 3.5 0 0 1-.2-7h1.2-1Zm0 3h5a.5.5 0 0 1 .09 1H5.5a.5.5 0 0 1-.09-1h.09Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span
            class="fui-MenuItem__content r1ls86vo"
          >
            Copy link
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="fui-MenuDivider ___3pxu8i0_ssaivc0 f8dz51a f14z66ap f1facbz3"
        role="presentation"
      />
      <div
        aria-labelledby="menu-groupr4"
        class="fui-MenuGroup"
        role="group"
      >
        <div
          class="fui-MenuItem rfoezjv"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="fui-MenuItem__icon ro9koqv"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 1a2 2 0 0 0-2 2v2.2c.32-.08.66-.15 1-.18V3a1 1 0 0 1 1-1h3v2.5c0 .83.67 1.5 1.5 1.5H13v7a1 1 0 0 1-1 1H9.74c-.31.38-.67.72-1.08 1H12a2 2 0 0 0 2-2V5.41c0-.4-.16-.78-.44-1.06l-2.91-2.91A1.5 1.5 0 0 0 9.59 1H6Zm6.8 4h-2.3a.5.5 0 0 1-.5-.5V2.2L12.8 5ZM10 10.5a4.5 4.5 0 1 1-9 0 4.5 4.5 0 0 1 9 0Zm-4-2a.5.5 0 0 0-1 0V10H3.5a.5.5 0 0 0 0 1H5v1.5a.5.5 0 0 0 1 0V11h1.5a.5.5 0 0 0 0-1H6V8.5Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span
            class="fui-MenuItem__content r1ls86vo"
          >
            New file
          </span>
        </div>
        <div
          class="fui-MenuItem rfoezjv"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="fui-MenuItem__icon ro9koqv"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 4.5V6h3.59a.5.5 0 0 0 .35-.15L7.29 4.5 5.94 3.15A.5.5 0 0 0 5.59 3H3.5C2.67 3 2 3.67 2 4.5Zm-1 0A2.5 2.5 0 0 1 3.5 2h2.09c.4 0 .78.16 1.06.44L8.2 4h4.29A2.5 2.5 0 0 1 15 6.5v.76a5.5 5.5 0 0 0-1-.66v-.1c0-.83-.67-1.5-1.5-1.5H8.2L6.66 6.56A1.5 1.5 0 0 1 5.59 7H2v4.5c0 .83.67 1.5 1.5 1.5h2.7c.1.35.24.68.4 1H3.5A2.5 2.5 0 0 1 1 11.5v-7Zm15 7a4.5 4.5 0 1 1-9 0 4.5 4.5 0 0 1 9 0Zm-4-2a.5.5 0 0 0-1 0V11H9.5a.5.5 0 0 0 0 1H11v1.5a.5.5 0 0 0 1 0V12h1.5a.5.5 0 0 0 0-1H12V9.5Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <span
            class="fui-MenuItem__content r1ls86vo"
          >
            New folder
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="fui-MenuDivider ___3pxu8i0_ssaivc0 f8dz51a f14z66ap f1facbz3"
        role="presentation"
      />
      <div
        aria-labelledby="menu-groupr5"
        class="fui-MenuGroup"
        role="group"
      >
        <div
          class="fui-MenuItem rfoezjv"
          role="menuitem"
          tabindex="0"
        >
          <span
            class="fui-MenuItem__icon ro9koqv"
          >
            <svg
              class="emotion-0"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.6924215,13.986674 C22.9997233,14.1908343 24,15.3216239 24,16.6862741 C24,17.7522298 23.3892094,18.6751163 22.4988002,19.1254521 C22.4321553,19.1592575 22.3639409,19.1904067 22.294157,19.2186583 L8.59404885,19.2186583 L8.59404885,19.2181754 C6.85476268,19.2073094 5.44797646,17.7943658 5.44797646,16.0524235 C5.44797646,14.3037201 6.86550796,12.8860679 8.61421133,12.8860679 C8.70222602,12.8860679 8.78903338,12.8906557 8.87511633,12.8976583 C8.87028699,12.8172498 8.86702719,12.7364791 8.86702719,12.6548633 C8.86702719,10.4662072 10.6413261,8.69190834 12.8299821,8.69190834 C14.3676434,8.69190834 15.6998164,9.5683125 16.3568479,10.8482079 C16.9002691,10.4918027 17.5498151,10.283779 18.2483789,10.283779 C20.1559675,10.283779 21.7023216,11.8301331 21.7023216,13.7377217 C21.7023216,13.8215107 21.6982167,13.9043338 21.6924215,13.986674 Z M8.18741857,12.2086324 C6.25507961,12.421727 4.74711875,14.0641849 4.74711875,16.0524235 C4.74711875,16.8619413 4.9992102,17.6150766 5.42841764,18.2378198 L3.27887919,18.2378198 C1.46799809,18.2378198 -1.77635684e-15,16.7698217 -1.77635684e-15,14.9589406 C-1.77635684e-15,13.2281057 1.34134869,11.8115401 3.04103428,11.6894786 C2.98428956,11.4324371 2.95350253,11.1656161 2.95350253,10.8915512 C2.95350253,8.85248384 4.60646427,7.19964283 6.64541087,7.19964283 C7.40977438,7.19964283 8.11980783,7.43181327 8.70886636,7.82963 C9.54542848,6.15252157 11.276867,5 13.2783862,5 C15.9223281,5 18.0967377,7.01021204 18.3569183,9.58557738 C18.3206982,9.58461151 18.2847197,9.58292125 18.2483789,9.58292125 C17.6798451,9.58292125 17.1232638,9.6982217 16.6079734,9.92012979 C15.7386926,8.71545137 14.3462736,7.99105063 12.8299821,7.99105063 C10.4089142,7.99105063 8.41282793,9.84563725 8.18741857,12.2086324 Z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="fui-MenuItem__content r1ls86vo"
          >
            Open with OneDrive
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the folder is undefined 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr6 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  />
</DocumentFragment>
`;
